R/utils-read.R

```diff
@@ -79,6 +79,75 @@ readFieldVisitMeasurementsQPoints <- function(reportObject){
   return(returnDf)
 }
 
+#' Read field visit readings
+#'
+#' @description Given a full report object, returns the field visit 
+#' readings formatted as a data frame
+#' @param reportObject the object representing the full report JSON
+readFieldVisitReadings <- function(reportObject,commentFlag){
+  visitReadings <- fetchFieldVisitReadings(reportObject)
+  requiredFields <- c('time')
+  returnDf <- data.frame(time=as.character(NA), party=as.character(NA), sublocation=as.character(NA), monitoringMethod=as.character(NA), value=as.character(NA), uncertainty=as.character(NA), estimatedTime=as.character(NA), comments=as.list(NA), associatedIvValue=as.character(NA), qualifiers=as.list(NA), associatedIvTime=as.character(NA), diffPeak=as.character(NA),stringsAsFactors=FALSE)
+
+  if(validateFetchedData(visitReadings, "Readings", requiredFields)){
+    time <- as.POSIXct(visitReadings[['time']])
+    party <- visitReadings[['party']]
+    sublocation <- visitReadings[['sublocation']]
+    monitoringMethod <- visitReadings[['monitoringMethod']]
+    value <- visitReadings[['value']]
+    uncertainty <- visitReadings[['uncertainty']]
+    estimatedTime <- visitReadings[['estimatedTime']] 
+    comments <- visitReadings[['comments']]
+    associatedIvValue <- visitReadings[['uncertainty']]
+    qualifiers <- sapply(readQualifiers(visitReadings[['associatedIvTime']], visitReadings[['associatedIvQualifiers']]), paste0)
+    associatedIvTime <- visitReadings[['associatedIvTime']]
+    diffPeak <- readIvDifference(visitReadings[['value']], visitReadings[['associatedIvValue']])
+    returnDf <- data.frame(time=time, party=party, sublocation=sublocation, monitoringMethod=monitoringMethod, value=value, uncertainty=uncertainty, estimatedTime=estimatedTime, comments=comments, associatedIvValue=associatedIvValue, qualifiers=qualifiers, associatedIvTime=associatedIvTime, diffPeak=diffPeak)
```

Comment:
currently when trying to test the handling of no qualifiers, this line results in an error:

`Error: Test failed: 'readFieldVisitReadings handles null qualifiers'
* arguments imply differing number of rows: 1, 0
1: repgen:::readFieldVisitReadings(reportObject, TRUE) at :24
2: data.frame(time = time, party = party, sublocation = sublocation, monitoringMethod = monitoringMethod, value = value, uncertainty = uncertainty, estimatedTime = estimatedTime, comments = comments, associatedIvValue = associatedIvValue, qualifiers = qualifiers, associatedIvTime = associatedIvTime, diffPeak = diffPeak) at D:/KMSData/git_repos/nawqa-nwis-data-check/repgen/R/utils-read.R:104
3: stop(gettextf("arguments imply differing number of rows: %s", paste(unique(nrows), collapse = ", ")), domain = NA)`